### ISSUE TITLE
Error: Cannot find module 'undici'

### ISSUE THREAD

**Issue by araines** [javajawa/check-registry-for-image#9]
First time user of your action, but got this error:

```
Run javajawa/check-registry-for-image@v3
node:internal/modules/cjs/loader:1148
  throw err;
  ^
Error: Cannot find module 'undici'
Require stack:
- /home/runner/work/_actions/javajawa/check-registry-for-image/v3/node_modules/@actions/http-client/lib/index.js
- /home/runner/work/_actions/javajawa/check-registry-for-image/v3/node_modules/@actions/core/lib/oidc-utils.js
- /home/runner/work/_actions/javajawa/check-registry-for-image/v3/node_modules/@actions/core/lib/core.js
- /home/runner/work/_actions/javajawa/check-registry-for-image/v3/check-tag.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Module._load (node:internal/modules/cjs/loader:98[6](https://github.com/araines/idrive-docker/actions/runs/11305497767/job/31445068277#step:8:7):27)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at require (node:internal/modules/helpers:1[7](https://github.com/araines/idrive-docker/actions/runs/11305497767/job/31445068277#step:8:8)9:18)
    at Object.<anonymous> (/home/runner/work/_actions/javajawa/check-registry-for-image/v3/node_modules/@actions/http-client/lib/index.js:41:1[8](https://github.com/araines/idrive-docker/actions/runs/11305497767/job/31445068277#step:8:9))
    at Module._compile (node:internal/modules/cjs/loader:1358:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)
    at Module._load (node:internal/modules/cjs/loader:1024:12)
    at Module.require (node:internal/modules/cjs/loader:1233:1[9](https://github.com/araines/idrive-docker/actions/runs/11305497767/job/31445068277#step:8:10)) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/runner/work/_actions/javajawa/check-registry-for-image/v3/node_modules/@actions/http-client/lib/index.js',
    '/home/runner/work/_actions/javajawa/check-registry-for-image/v3/node_modules/@actions/core/lib/oidc-utils.js',
    '/home/runner/work/_actions/javajawa/check-registry-for-image/v3/node_modules/@actions/core/lib/core.js',
    '/home/runner/work/_actions/javajawa/check-registry-for-image/v3/check-tag.js'
  ]
}
Node.js v20.[13](https://github.com/araines/idrive-docker/actions/runs/11305497767/job/31445068277#step:8:14).1
```

**Comment by javajawa** [comment]
That's...odd. Nothing in this repo's code directly uses `undici` which makes me suspect that something has changed in a dependency in a way the frozen modules don't handle?

I have created https://github.com/javajawa/check-registry-for-image/pull/10 which is just a updated rebuild of all dependencies -- could you see if that works better for you (run with `uses: javajawa/check-registry-for-image@update-dependencies` in your CI pipeline)

**Comment by araines** [comment]
I'm still getting the same problem.  It looks like it is coming in via [@actions/http-client](https://github.com/javajawa/check-registry-for-image/blob/mainline/package-lock.json#L1463C5-L1463C30) looking at the lock file:

```
    "@actions/http-client": {
      "version": "2.2.1",
      "resolved": "https://registry.npmjs.org/@actions/http-client/-/http-client-2.2.1.tgz",
      "integrity": "sha512-KhC/cZsq7f8I4LfZSJKgCvEwfkE8o1538VoBeoGzokVLLnbFDEAdFD3UhoMklxo2un9NJVBdANOresx7vTHlHw==",
      "requires": {
        "tunnel": "^0.0.6",
        "undici": "^5.25.4"
      }
    },
```

But I don't see it in the `node_modules` directory.  I've not actually built a GitHub action with JS myself, but I'm wondering whether the `node_modules` directory should actually be checked in?

**Comment by javajawa** [comment]
Somewhat frustratingly, yes `node_modules` [has to be checked in](https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github) -- the action runner grabs the artifact and executes the code with 0 other setup.

Naturally, I have `node_modules` in my global gitignore, so I think that every rebuild of the dependencies I've done recently has failed to add any new dependencies! I've just pushed another new version to the same branch which includes 100+ new files, hopefully that will work better.

(I should set up proper CI and renovate for this project, because I no longer use it myself...)

**Comment by araines** [comment]
Perfect, that sorted it!  I've got an authentication error now, but that is a "me" problem!  Thanks for getting this sorted so quickly.

**Comment by javajawa** [comment]
Thanks for checking out the action!
I'll get it properly tagged up as `v3` and then close this issue.

**Comment by javajawa** [comment]
This is now available as `v3` (or `v3.1` if you desire to pin to a fixed point).
